python/lsst/daf/butler/_butler.py

```diff
@@ -1670,6 +1670,78 @@ def doImport(importStream: TextIO) -> None:
         else:
             doImport(filename)
 
+    @transactional
+    def transfer_from(self, source_butler: Butler, collection: str,
+                      transfer: str = "auto") -> List[DatasetRef]:
+        """Transfer datasets from a run in another Butler.
+
+        Parameters
+        ----------
+        source_butler : `Butler`
+            Butler from which the datasets are to be transferred.
+        collection : `str`
+            Name of the collection to transfer.
+        transfer : `str`, optional
+            Transfer mode passed to `~lsst.daf.butler.Datastore.transfer_from`.
+
+        Returns
+        -------
+        refs : `list` of `DatasetRef`
+            The refs added to this Butler.
+
+        Notes
+        -----
+        Requires that any dimension definitions are already present in the
+        receiving Butler. The datastore artifact has to exist for a transfer
+        to be made but non-existence is not an error.
+
+        The source collection will be reconstructed in this butler using
+        the same names. If a dataset already exists in a RUN collection
+        of the same name no transfer will occur and this will not be an error
+        unless the DatasetRef uses UUID and there is a UUID mismatch.
```

Comment:
I think for importDatasets if the ref already exists in the repo it silently skips it. If somehow the UUID does not match the ref in the registry but does match the DataId/DatasetType then it will complain won't it? We can imagine a scenario like this if people do a transfer, then prune their copy and rerun and forget and transfer again. Hence the comment about it being fine if the UUID is used and matches but not if the UUID differs.